src/coreclr/jit/hwintrinsic.h

```diff
@@ -312,6 +312,106 @@ enum class FloatRoundingMode : uint8_t
     // _MM_FROUND_NO_EXC
     NoException = 0x08,
 };
+
+enum class TernaryLogicUseFlags : uint8_t
+{
+    // Indicates no flags are present
+    None = 0,
+
+    // Indicates the ternary logic uses A
+    A = 1 << 0,
+
+    // Indicates the ternary logic uses B
+    B = 1 << 1,
+
+    // Indicates the ternary logic uses C
+    C = 1 << 2,
+
+    // Indicates the ternary logic uses A and B
+    AB = (A | B),
+
+    // Indicates the ternary logic uses A and C
+    AC = (A | C),
+
+    // Indicates the ternary logic uses B and C
+    BC = (B | C),
+
+    // Indicates the ternary logic uses A, B, and C
+    ABC = (A | B | C),
+};
+
+enum class TernaryLogicOperKind : uint8_t
+{
+    // Indicates no operation is done
+    None = 0,
+
+    // value
+    Select = 1,
+
+    // constant true (1)
+    True = 2,
+
+    // constant false (0)
+    False = 3,
+
+    // ~value
+    Not = 4,
+
+    // left & right
+    And = 5,
+
+    // ~(left & right)
+    Nand = 6,
+
+    // left | right
+    Or = 7,
+
+    // ~(left | right)
+    Nor = 8,
+
+    // left ^ right
+    Xor = 9,
+
+    // ~(left ^ right)
+    Xnor = 10,
+
+    // cond ? left : right
+    Cond = 11,
```

Comment:
looking forward to seeing all these to be optimizations for corresponding patterns